File: Selection_Sort/selection_sort.c
/*
    Selection Sort
    Time Complexity
        Best Case Scenario  : O(n^2)
        Worst Case Scenario : O(n^2)
*/
#include <stdio.h>
void swap(int *a,int *b)
{
    int temp = *a;
    *a=*b;
    *b=temp;
}
void selection_sort(int arr[],int n)
{
    int i,j,min_index;
    for(i=0;i<n-1;i++)
    {
        min_index=i;
        for(j=i+1;j<n;j++)
        {
            if(arr[j]<arr[min_index])
            {
                min_index=j;
            }
        }
        swap(&arr[min_index],&arr[i]);
    }
}
int main()
{
    int arr[]={1,-10,-30,20,2,11,15};
    int n=sizeof(arr)/sizeof(arr[0]),i;
    selection_sort(arr,n);
    printf("Array after selection sort : ");
    for(i=0;i<n;i++)
    {
        printf("%d ",arr[i]);
    }
    printf("\n");
    return 0;
}

File: Sorting-Algorithms/Selection_Sort/Java/SelectionSort.java
/*
    Selection Sort
    Time Complexity
        Best Case Scenario  : O(n^2)
        Average Case Scenario : O(n^2)
        Worst Case Scenario : O(n^2)
*/
import java.util.*;
public class SelectionSort
{
    static void selection_sort(int A[],int n)
    {
        for(int i=0;i<n-1;i++)
        {
            int min=i;
            for(int j=i+1;j<n;j++)
            {
                if(A[j]<A[min])
                {
                    min=j;
                }
            }
            int temp=A[min];
            A[min]=A[i];
            A[i]=temp;
        }
    }
    public static void main(String args[])
    {
        int arr[]={1,-10,-30,20,2,11,15};
        int n=arr.length,i;
        selection_sort(arr,n);
        System.out.print("Array after Selection sort : ");
        for(i=0;i<n;i++)
        {
            System.out.print(arr[i]+" ");
        }
        System.out.println();
    }
}
